File: November2021/FindAllNumbersDisappearedinanArray.java
// O(n)
class Solution {
	public List<Integer> findDisappearedNumbers(int[] nums) {
		List<Integer> ans = new ArrayList<>();
		for(int i=0;i<nums.length;i++){
			int el = Math.abs(nums[i])-1;
			if(nums[el]>0){
				nums[el] = -nums[el];
			}
		}

		for(int i=0;i<nums.length;i++){

			if(nums[i]>0){
				ans.add(i+1);
			}
		}
		return ans;
	}
}

File: November2021/C++/Find_All_Numbers_Disappeared_in_an_Array.cpp
// author:shyam2520
// https://leetcode.com/problems/find-all-numbers-disappeared-in-an-array/
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
      for(int i=0;i<nums.size();i++)
      {
          // cout<<nums[nums[i]-1]<<endl;
          if(nums[abs(nums[i])-1]<0)
              continue;
          else
              nums[abs(nums[i])-1]=-(nums[abs(nums[i])-1]);
      }
      vector<int> res;
      for(int i=0;i<nums.size();i++)
          // cout<<i<<" "<<nums[i]<<endl;
          if(nums[i]>0) res.push_back(i+1);
      return res;
    }
};